test/agency_test.exs

```diff
@@ -6,8 +6,10 @@ defmodule Agency.Test do
     {:ok, pid1} = TestAgency1.start_link()
     {:ok, pid2} = TestAgency2.start_link()
     {:ok, pid3} = TestAgency3.start_link()
+    {:ok, pid4} = TestAgency4.start_link(name: TA4)
```

Comment:
It would be interesting to verify that you can actually start two different agencies from the same module, and they will be independent.

I'm 180% sure it works, I just would like to see it in a test :)